### ISSUE TITLE
memoryStorage.setItem() won't overwrite existing value

### ISSUE THREAD

**Issue by drmercer** [TechnologyAdvice/local-sync#12]
I was looking through the source, and I found what looks like a bug. The code for setItem() in memoryStorage does nothing if a value is already set for that key, but it should update it according to the doc comment.
```javascript
setItem(key, value) {
    if (typeof value === 'undefined') {
      this.removeItem(key)
    } else {
      // Here's the problem:
      if (!cache[key]) {
        cache[key] = value.toString()
        length++
      }
    }
  },
```

**Comment by levithomason** [comment]
Yep, bug.  I don't have rights to this repo any longer, otherwise, I'd fix it 😄 What is actually needed here is a simple test that compares the actual local storage behavior to the memory version.

@davezuko or @TomFrost this is another module [I'd written solo](https://github.com/TechnologyAdvice/local-sync/graphs/contributors) and don't mind maintaining if transferred.  If not, that's cool too.  As a heads up, this module is used in a few prod apps and some (limited) users will be hitting this bug in certain places.

**Comment by levithomason** [comment]
Released in `local-sync@2.0.3`
